File: C#高级编程(第7版)源代码/502259 Code CH53/CSharp/Singleton.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace CSharp
{
    public class SomeData
    {
    }

    public class Singleton
    {
        private static SomeData d = null;
        public static SomeData GetObject()
        {
            if (d == null)
            {
                d = new SomeData();
            }
            return d;
        }
    }
}

File: C#高级编程(第7版)源代码/502259 Code CH53/VisualBasic/Singleton.vb
Imports System
Imports System.Collections.Generic
Imports Message = System.Messaging

Namespace Wrox.ProCSharp.Languages
End Namespace


Public Class SomeData

End Class

Public Class Singleton
    Private Shared d As SomeData

    Public Shared Function GetData() As SomeData
        If d Is Nothing Then
            d = New SomeData()
        End If
        Return d

    End Function

End Class
